src/topology/algebra/ordered.lean

```diff
@@ -297,6 +297,33 @@ begin
   exact ⟨x, le_antisymm hfg hgf⟩
 end
 
+lemma intermediate_value_univ₂_cl_snd {s : set γ} (hs : is_preconnected s)
+  {a b : γ} (ha : a ∈ s) (hb : b ∈ closure s) {f g : γ → α} (hf : continuous f)
+  (hg : continuous g) (ha' : f a ≤ g a) (hb' : g b < f b) : ∃ x ∈ s, f x = g x :=
```

Comment:
The first version shouldn't mention a set `s` at all (note how it's done for the lemma just above) -- hence the `univ` in the name.  I think this is the most general version with no set `s`:
```lean
lemma intermediate_value_univ₂_eventually [preconnected_space γ] {a : γ} {b : filter γ} [ne_bot b]
  {f g : γ → α} (hf : continuous f) (hg : continuous g) (ha : f a ≤ g a)
  (hb : ∀ᶠ x in b, g x ≤ f x) :
  ∃ x, f x = g x :=
let ⟨c, hc⟩ := hb.frequently.exists in intermediate_value_univ₂ hf hg ha hc
```